### PR TITLE
Install devDependencies to migrate Storybook, Typescript and Jest to current versions

### PR DESCRIPTION
**Pull request by millipede-cpu**
- Install missing devDependencies allowing unit tests to run locally.
- Upgrade typescript to current version of `"5.2.2"`
- Fix failing e2e test `nextbutton.test.ts`
- Remove `headless: false` that opens chromium browsers locally when running playwright e2e tests
- [Migrate Storybook](https://storybook.js.org/docs/react/migration-guide) to current version of `7.4.4`

<p align="center">
Jest unit tests passing with added devDepencies 🥳 
Screenshot 2023-09-24 at 17 28 05

</p>

<p align="center">
Playwright tests pass and run much faster without `slowMo` mode and `headless: false` 🦥 
Screenshot 2023-09-24 at 17 13 53

</p>

Co-authored-by: Peter <PJSalter@users.noreply.github.com>
Co-authored-by: Milly <millipede-cpu@users.noreply.github.com>

### CODE REVIEW

**Comment by PJSalter** [comment]
> * Install missing devDependencies allowing unit tests to run locally.
> * Upgrade typescript to current version of `"5.2.2"`
> * Fix failing e2e test `nextbutton.test.ts`
> * Remove `headless: false` that opens chromium browsers locally when running playwright e2e tests
> 
> Jest unit tests passing with added devDepencies 🥳 Screenshot 2023-09-24 at 17 28 05
> 
> Playwright tests pass and run much faster without `slowMo` mode and `headless: false` 🦥 Screenshot 2023-09-24 at 17 13 53
> 
> Co-authored-by: Peter [PJSalter@users.noreply.github.com](mailto:PJSalter@users.noreply.github.com) Co-authored-by: Milly [millipede-cpu@users.noreply.github.com](mailto:millipede-cpu@users.noreply.github.com)

Yay this is amazing, thank you for these improvements to get all the Unit and end2end tests passing 🤩🥳

File: e2e-tests/addJoke.test.ts [hunk 1]
```diff
@@ -2,10 +2,7 @@ import { chromium, expect, test } from "@playwright/test";
 
 test("Typing a joke and a pun and clicking the seperate buttons should render the text on the homepage", async () => {
   // Simualte a chromium browser with an attribute to slow the running time down by 3 seconds so that you can visually see the test going through each step.
-  const browser = await chromium.launch({
```
Comment (PJSalter) [inline e2e-tests/addJoke.test.ts:5]: Thank you for removing slowMo and headless false, the tests work so much better now, running smoother and faster 🧪🤩

File: e2e-tests/nextbutton.test.ts [hunk 2]
```diff
@@ -5,87 +5,50 @@ test.describe("Button component is visible on multiple pages", () => {
...
-    await page.goto("http://localhost:5173/");
-    page = await browser.newPage();
-    await page.goto("http://localhost:5173/joke-randomiser");
-    page = await browser.newPage();
-    await page.goto("http://localhost:5173/card-flip-game");
-    page = await browser.newPage();
-    await page.goto("http://localhost:5173/rate-joke");
-    page = await browser.newPage();
-    await page.goto("http://localhost:5173/joke-fetch");
+    await page.goto("http://localhost:5173");
```
Comment (PJSalter) [inline e2e-tests/nextbutton.test.ts:17]: a lot more neater 😁🙌

File: e2e-tests/nextbutton.test.ts [hunk 3]
```diff
@@ -5,87 +5,50 @@ test.describe("Button component is visible on multiple pages", () => {
...
   });
 
-  test.afterEach(async () => {
-    await page.close();
-  });
+  // Define a reusable function to navigate to pages and check buttons
+  async function navigateToPageAndCheckButton(pageUrl) {
```
Comment (PJSalter) [inline e2e-tests/nextbutton.test.ts:21]: This is great so does this function allow you to test each next button clicks on every page?

File: e2e-tests/nextbutton.test.ts [hunk 4]
```diff
@@ -5,87 +5,50 @@ test.describe("Button component is visible on multiple pages", () => {
...
   test("should be visible on the home page", async () => {
-    try {
-      await expect(page.locator('button[class="next-button"]')).toBeVisible();
-    } catch (e) {
-      console.error(e);
+    // Check the home page
+    await expect(page.getByTestId("next-button")).toBeVisible();
+  });
+
+  test("should navigate through pages and check the button", async () => {
```
Comment (PJSalter) [inline e2e-tests/nextbutton.test.ts:31]: Amazing 🥳🤩 I like the use of comments, it describes the test very clearly 🧪✨

File: src/NextButton.tsx [hunk 5]
```diff
@@ -21,6 +21,7 @@ export default function BackButtonWrapper({ to }: Props): JSX.Element {
   return (
     <Button
       className="next-button"
+      data-testid="next-button"
```
Comment (PJSalter) [inline src/NextButton.tsx:24]: must cleaner way for the test for the next-button to be picked up, great work 🙌✨

File: package.json [hunk 6]
```diff
@@ -38,14 +39,16 @@
...
+    "@types/react-dom": "^18.2.7",
     "babel-loader": "^8.3.0",
     "babel-plugin-styled-components": "^2.1.4",
     "babel-plugin-transform-react-jsx": "^6.24.1",
     "jest": "^29.7.0",
     "react-test-renderer": "^18.2.0",
     "ts-jest": "^29.1.1",
     "ts-node": "^10.9.1",
-    "typescript": "^5.0.4",
+    "typescript": "^5.2.2",
```
Comment (PJSalter) [inline package.json:51]: Thank you for updating and installing the latest typeScript and including the dev dependencies that we need for our application to run all tests smoothly 🧪🙌🤩🥳✨⭐️

**Review by PJSalter — APPROVED** [review]

File: e2e-tests/nextbutton.test.ts (hunk 3 above)
Comment (millipede-cpu) [reply e2e-tests/nextbutton.test.ts:21]: Yes, it removes a lot of lines of repeated code, where we were adding a test for each page separately, where now the `navigateToPageAndCheckButton` function will run asynchronously aka not in order and iterates through each page as a combined test, which prevents it from timing out and failing also 🧪

**Pull request merged**